R/data-combined.R

```diff
@@ -0,0 +1,625 @@
+#' `SimulationResults` object.
+#' - `$addDataSets()`: In the context of this method, a list of strings
+#' assigning new names to the list of instances of the `DataSet` object.
+#'
+#' Note that the datasets whose names you wish to not change should be specified
+#' as `NULL` in the list.
+#' @param xOffsets,yOffsets,xScaleFactors,yScaleFactors Either a numeric scalar
+#'   or a list of numeric quantities specifying offsets and scale factors to
+#'   apply to raw values. The default offset is `0`, while default scale factor
+#'   is `1`, i.e., the data will not be modified. If a list is specified, it
+#'   should be the same length as `names` argument.
+#'
+#' @import tidyr
+#' @importFrom rlang ":=" "%||%"
+#' @importFrom ospsuite.utils Printable
+#' @importFrom ospsuite.utils enum enumGetValue enumHasKey enumKeys enumPut enumRemove enumValues
+#' @importFrom ospsuite.utils formatNumerics
+#' @importFrom ospsuite.utils getEnumKey
+#' @importFrom ospsuite.utils hasUniqueValues
+#' @importFrom ospsuite.utils ifNotNull
+#' @importFrom ospsuite.utils isFileExtension isIncluded isOfLength isOfType isSameLength
+#' @importFrom ospsuite.utils toList
+#' @importFrom ospsuite.utils validateEnumValue
+#' @importFrom ospsuite.utils validateIsIncluded
+#' @importFrom ospsuite.utils validateIsInteger
+#' @importFrom ospsuite.utils validateIsLogical
+#' @importFrom ospsuite.utils validateIsNumeric
+#' @importFrom ospsuite.utils validateIsOfLength
+#' @importFrom ospsuite.utils validateIsOfType
+#' @importFrom ospsuite.utils validateIsSameLength
+#' @importFrom ospsuite.utils validateIsString
+#' @importFrom ospsuite.utils validatePathIsAbsolute
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSets(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
+
+  public = list(
+
+    ## constructor method ---------------
+
+    #' @description
+    #' Initialize a new instance of the class.
+    #' The constructor is empty because we prefer that the users add
+    #' `SimulationResults` and `DataSet` objects separately and individually.
+    #' @return A new instance of `DataCombined` object.
+
+    # empty constructor
+    initialize = function() {
+    },
+
+    ## setter methods ---------------
+
+    #' @description
+    #' Add simulated data.
+    #' @return `DataCombined` object containing simulated data.
+
+    addSimulationResults = function(simulationResults,
```

Comment:
I would prefer to have the order of arguments similar to that in `getOutputValues`, namely

`simulationResults, quantitiesOrPaths, population, individualIds, names, groups`